scrapy/http/response/text.py

```diff
@@ -116,3 +120,57 @@ def xpath(self, query, **kwargs):
 
     def css(self, query):
         return self.selector.css(query)
+
+    def follow(self, url, callback=None, method='GET', headers=None, body=None,
+               cookies=None, meta=None, encoding=None, priority=0,
+               dont_filter=False, errback=None):
+        # type: (...) -> Request
+        """
+        Return a scrapy.Request instance to follow a link ``url``.
+
+        ``url`` can be:
+
+        * absolute URL;
+        * relative URL;
+        * scrapy.link.Link object (e.g. a link extractor result);
+        * attribute Selector (not SelectorList) - e.g.
+          ``response.css('a::attr(href)')[0]`` or
+          ``response.xpath('//img/@src')[0]``.
+        * a Selector for ``<a>`` element, e.g.
+          ``response.css('a.my_link')[0]``.
+        """
+        if isinstance(url, Link):
+            url = url.url
+        elif isinstance(url, parsel.Selector):
+            url = _url_from_selector(url).strip()
```

Comment:
this could benefit from `trim_href_attribute` function from https://github.com/scrapy/scrapy/pull/2547